### PR TITLE
Basic rich presence

### PR DESCRIPTION
**Pull request by Shosoul**
Adds rich presence to the game if discord-rpc.dll in the installer folder.

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -83,14 +95,21 @@ function love.update(dt)
...
+    error(err .. "\n" .. debug.traceback(mainloop) .. "\n" .. system_info)
   end
   if server_queue and server_queue:size() > 0 then
     logger.trace("Queue Size: " .. server_queue:size() .. " Data:" .. server_queue:to_short_string())
   end
   this_frame_messages = {}
 
   update_music()
+  if discordRPC then
+    if nextPresenceUpdate < love.timer.getTime() then
```
Comment (JamesVanBoxtel) [inline main.lua:107]: Lets not poll like this throughout the game, instead lets just update immediately whenever the rich_presence variable changes

**Comment by Shosoul** [comment]
Ok, this should be better.

File: main.lua (hunk 1 above)
Comment (Shosoul) [reply main.lua]: Solved.

File: match.lua [hunk 2]
```diff
@@ -13,7 +12,20 @@ Match =
     GAME.droppedFrames = 0
     self.supportsPause = true
     self.attackEngine = nil
-    self.current_music_is_casual = true 
+    self.current_music_is_casual = true
+    self.startTimestamp = os.time(os.date("*t"))
+    if P2 or mode == "vs" then
+      print(mode)
```
Comment (JamesVanBoxtel) [inline match.lua:18]:
```suggestion
```

**Review by JamesVanBoxtel — APPROVED** [review]
please delete the print. Later we should move this out of match so it works with replays etc.

File: match.lua (hunk 2 above)
Comment (Shosoul) [reply match.lua]: Thanks. I forgot to take that out while debugging.

**Pull request merged**